Order unordered querysets by primary key

SupImDos opened the pull request:
## Description

As per #714, types for models with no default `Meta.ordering` can result in non-deterministic results. This PR adds a `.order_by("pk")` to unordered QuerySets for any list, connection, paginated or optional type. It also adds some extra fields to the `tests/projects/schema.py` GraphQL schema to enable the testing of more combinations of unordered models / types / fields.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

Fix #714

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

## Summary by Sourcery

Orders unordered QuerySets by primary key to ensure deterministic results for list, connection, paginated or optional types.

Bug Fixes:
- Fixes a bug where models with no default `Meta.ordering` can result in non-deterministic results.

Tests:
- Adds tests to verify that queries on list fields and nested list fields with no ordering have an `ORDER BY` clause.

## Summary by Sourcery

Ensures deterministic ordering of query results by ordering unordered QuerySets by their primary key. This change affects list, connection, paginated, and optional types, guaranteeing consistent results when no explicit ordering is defined in the model's Meta class.

Bug Fixes:
- Fixes non-deterministic results for models without default ordering in Meta, ensuring consistent query results.

Enhancements:
- Adds primary key ordering to unordered QuerySets for list, connection, paginated, or optional types.

Tests:
- Adds tests to verify that queries on list fields and nested list fields with no ordering have an `ORDER BY` clause.

bellini666 commented:
Hi @SupImDos ,

I like the solution :)

tip: rebase this on top of `main` as I fixed the current typing issues which broke in this PR

SupImDos commented:
> Hi @SupImDos ,
> 
> I like the solution :)
> 
> tip: rebase this on top of `main` as I fixed the current typing issues which broke in this PR

Thanks @bellini666! I've rebased and fleshed out the unit tests, but it looks like there are still some typing errors for some reason :shrug:

bellini666 commented:
> > Hi @SupImDos ,
> > I like the solution :)
> > tip: rebase this on top of `main` as I fixed the current typing issues which broke in this PR
> 
> Thanks @bellini666! I've rebased and fleshed out the unit tests, but it looks like there are still some typing errors for some reason 🤷

It is because of a revert from pyright lol

Will merge this and fix the typing issues in a following commit
